x/wasm/keeper/api.go

```diff
@@ -22,25 +24,32 @@ const (
 var (
 	costHumanize            = DefaultGasCostHumanAddress * types.DefaultGasMultiplier
 	costCanonical           = DefaultGasCostCanonicalAddress * types.DefaultGasMultiplier
+	costValidate            = DefaultGasCostCanonicalAddress * types.DefaultGasMultiplier
 	costJSONDeserialization = wasmvmtypes.UFraction{
 		Numerator:   DefaultDeserializationCostPerByte * types.DefaultGasMultiplier,
 		Denominator: 1,
 	}
 )
 
-func humanAddress(canon []byte) (string, uint64, error) {
+func humanizeAddress(canon []byte) (string, uint64, error) {
 	if err := sdk.VerifyAddressFormat(canon); err != nil {
 		return "", costHumanize, err
 	}
 	return sdk.AccAddress(canon).String(), costHumanize, nil
 }
 
-func canonicalAddress(human string) ([]byte, uint64, error) {
+func canonicalizeAddress(human string) ([]byte, uint64, error) {
 	bz, err := sdk.AccAddressFromBech32(human)
 	return bz, costCanonical, err
 }
 
+func validateAddress(human string) (uint64, error) {
+	_, err := sdk.AccAddressFromBech32(human)
+	return costValidate, err
```

Comment:
This implementation does not yet check that the address is normalized since bech32 allows both uppercase and lower case inputs.

For a generic normalization check we need to convert it back and and check if we get the same result as the input.